packages/buidler-core/src/internal/core/runtime-environment.ts

```diff
@@ -309,10 +309,6 @@ export class Environment implements BuidlerRuntimeEnvironment {
     argumentValue: any
   ) {
     const { name: paramName, type, isVariadic } = paramDefinition;
-    if (type === undefined || type.validate === undefined) {
-      // no type or no validate() method defined, just skip validation.
-      return;
-    }
```

Comment:
I'm curious about why this was removed.

Reply:
`validate` wasn't present in the initial version of builder, so we added it as optional to make the change backward compatible. Now it's not optional anymore.